src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,206 @@
+        }catch(IOException e){
+            e.printStackTrace();
+        }
+    }
+
+    public static void storeTextAndList() throws DukeException {
+        String command;
+        Scanner in = new Scanner(System.in);
+        command = in.nextLine();
+        while (!command.equals("bye")) {
+            System.out.println(horizontalLine);
+            try {
+                if (command.equals("list")) {
+                    listTask();
+                } else if (command.contains("done")) {
+                    int index = Integer.parseInt(command.substring(5));
+                    if(index > thingsCounted){
+                        throw new DukeException();
+                    }
+                    markTaskAsDone(index);
+                } else if (command.contains("todo")) {
+                    addTodoTask(command);
+                } else if (command.contains("deadline")) {
+                    addDeadlineTask(command);
+                } else if (command.contains("event")) {
+                    addEventTask(command);
+                } else if (command.contains("delete")){
+                    int index = Integer.parseInt(command.substring(7));
+                    if(index > thingsCounted){
+                        throw new DukeException();
+                    }
+                    deleteItem(index);
+                } else {
+                    throw new DukeException();
+                }
+            }catch (DukeException e) {
+                dealWithException(command);
+            }
+                System.out.println(horizontalLine);
+                command = in.nextLine();
+            }
+            System.out.println(horizontalLine);
+            System.out.println("Bye. Hope to see you again soon");
+            System.out.println(horizontalLine);
+    }
+
+    public static void dealWithException(String text){
+        if (text.equals("todo")) {
+            System.out.println("☹ OOPS!!! The description of a todo cannot be empty.");
+        } else if (text.equals("deadline")) {
+            System.out.println("☹ OOPS!!! The description of a deadline cannot be empty.");
+        } else if (text.equals("event")) {
+            System.out.println("☹ OOPS!!! The description of a event cannot be empty.");
+        } else if(text.contains("done")){
+            System.out.println("☹ OOPS!!! The index is out of bound");
+        } else if(text.contains("delete")){
+            System.out.println("☹ OOPS!!! The index is out of bound");
+        } else {
+            System.out.println(("☹ OOPS!!! I'm sorry, but I don't know what that means :-("));
+        }
+    }
+
+    public static void addTodoTask(String text) throws DukeException {
+        String todoDescription;
+        if(text.equals("todo")){
+            throw new DukeException();
+        }
+        todoDescription = text.substring(5);
+        Task task = new Todo(todoDescription);
+        //tasks[thingsCounted] = task;
```

Comment:
Do remove commented code if they are ultimately not used in your final submission.